src/main/java/seedu/address/model/event/Event.java

```diff
@@ -0,0 +1,92 @@
+package seedu.address.model.event;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Objects;
+
+import seedu.address.commons.util.ToStringBuilder;
+
+/**
+ * Represents an Event in the address book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Event {
+
+    // Identity fields
+    private final EventName eventName;
+
+    // Data fields
+    private final Location location;
+    private final EventDate eventDate;
+
+    /**
+     * Every field must be present and not null.
+     */
+    public Event(EventName eventName, EventDate eventDate, Location location) {
+        requireAllNonNull(eventName, eventDate, location);
+        this.eventName = eventName;
+        this.eventDate = eventDate;
+        this.location = location;
+    }
+
+    public EventName getName() {
+        return eventName;
```

Comment:
I would advocate for adding the this keyword when referring to variables within the class to make a clearer distinction!

Reply:
Understood, I will use `this`